src/java.desktop/share/classes/javax/swing/plaf/metal/MetalBorders.java

```diff
@@ -241,41 +263,90 @@ public void paintBorder(Component c, Graphics g, int x, int y,
                 shadow = MetalLookAndFeel.getControlInfo();
             }
 
-              g.setColor(background);
-              // Draw outermost lines
-              g.drawLine( 1, 0, w-2, 0);
-              g.drawLine( 0, 1, 0, h-2);
-              g.drawLine( w-1, 1, w-1, h-2);
-              g.drawLine( 1, h-1, w-2, h-1);
+            Graphics2D g2d = (Graphics2D) g;
```

Comment:
It could be a DebugGraphics which is not a Graphics2D

Reply:
If the code will be refactored please double-check that the blind cast will not be used, "g" could DebugGraphics which is not a Graphics2D